### PR TITLE
Colour utility

### PR DESCRIPTION
**Pull request by nastiatikk**
Status: **Ready for Review**
Reviewers: @ry5n @kpeatt @avelinet @cole-sanderson @yourpalsonja @jeffkamo @mlworks 

## Changes
`u-text-accent`, `u-text-success` and `u-text-error` utilities added to `_text.scss`

## Proof of existence 
Sometimes we need to add just text colour to a word. With component approach we don't have many template files and sometimes we create them for just one line. This is where colour utility could help us.

I couldn't find a way to avoid using `$text__accent` variables. Is there a way to use Vellum variables without it?

## Todos:
- [x] Designer +1 (@jeffkamo)
- [x] Designer +1 (@ry5n)

### CODE REVIEW

File: dist/utils/text/_text.scss [hunk 1]
```diff
@@ -1,6 +1,10 @@
 // Text utils
 // ===
 
+$text__accent: if(variable-exists(accent-colot), $accent-color, blue) !default;
```
Comment (jeffkamo) [inline dist/utils/text/_text.scss]: typo: `accent-color` not `accent-colot`

File: dist/utils/text/_text.scss [hunk 2]
```diff
@@ -50,3 +54,24 @@
...
 }
+
+
+// Colour accents
+// ---
+//
+// When we need to mark text with colour
+
+.u-text-accent,
+%u-text-accent {
```
Comment (jeffkamo) [inline dist/utils/text/_text.scss]: @ry5n didn't we eventually decide not to add placeholder selectors to our utility classes? Or am I misremembering?

Comment (ry5n) [inline dist/utils/text/_text.scss]: We added them in. See https://github.com/mobify/stencil/pull/104

**Comment by ry5n** [comment]
I’m not a big fan of adding utils for text colors. I’m just not convinced that enough projects need them. (You can always add what you need in the project, since these are so simple.)

**Comment by jeffkamo** [comment]
I dunno, I've found myself in scenarios where all I needed to do was change the color of some text. Options being:

a. Create a new component (seems a bit heavy handed for just a color)
b. Add to an existing component (may or may not be valid depending on circumstances)
c. Add some page styles (higher specificity bandaid)
d. Use a utility class (it's another utility class, doesn't make sense in places that could change)

So ya, I dunno... I'm not totally against it... I understand why some people are against it... maybe I'm just lazy haha.

**Comment by nastiatikk** [comment]
Can you explain more your position? Why you're not a fan of adding utils for text colours?

We have utilities to avoid creating 1-property templates, or to avoid creating a component that we use from project to project. We have a lot of very simple utils like margins, all-caps, floats, visibility, text-align. Colours look like another one simple util for those who needs it.

**Comment by ry5n** [comment]
I should clarify: I am definitely for creating utilities for this. But I am not sure they belong in our official utilities, mostly because the colours you need will vary between projects. The utils we have here are invariant: all-caps, text-align are useful on any project.

I’m not totally opposed to them if others feel these ones would be often used.

**Comment by avelinet** [comment]
@ry5n i think the 3 colours @nastiatikk is proposing are pretty standard and used on most projects. At the very least the success and error ones.

**Comment by nastiatikk** [comment]
@avelinet is right, probably accent-colour isn't wide used and we can remove it. But `success` and `error` colours are used in the single selected words often enough to put them to the util.

**Comment by ry5n** [comment]
OK, cool. I don’t think we’ve done config variables before for utilities. Wondering if we should add a `u-` variable prefix to distinguish this from a hypothetical Text component:

```scss
$u-text__success: …;
```

File: tests/visual/utils/text/text.css [hunk 3]
```diff
@@ -22,3 +22,15 @@
 .u-align-baseline {
   vertical-align: baseline !important;
 }
+
+.u-text-accent {
+  color: blue;
+}
```
Comment (ry5n) [inline tests/visual/utils/text/text.css]: Need to make sure we re-compile and commit the css for gh-pages since it still has the accent output right now.

**Comment by jeffkamo** [comment]
Thanks for clarifying @ry5n. Sounds like we've come to a reasonable concensus. Once that CSS is recompiled and commited, :+1:

**Comment by ry5n** [comment]
Also want to get thoughts from everyone on the variable name suggestion above.

**Comment by jeffkamo** [comment]
it's probably a good idea to namespace the utility vars, otherwise they look like they are referencing components.

**Comment by ry5n** [comment]
Agreed.

**Comment by nastiatikk** [comment]
`u-text__something` is a good idea to separate them from component variables

**Comment by jeffkamo** [comment]
:+1:

**Comment by ry5n** [comment]
+1

**Pull request merged**